react/components/MinicartFreeshipping.tsx

```diff
@@ -11,27 +12,46 @@ import AppSettings from './minicartbarSettings.graphql'
 interface SettingsProps {
   settings: Settings
 }
+
 interface Settings {
   freeShippingAmount: number
+  isMultiBindingStore: boolean
+  freeShippingAmounts: [FreeShippingAmounts]
+}
+
+interface FreeShippingAmounts {
+  bindingId: string
+  freeShippingAmount: number
 }
 
 type ValueTypes = 'Discounts' | 'Items'
 
 const MinimumFreightValue: FunctionComponent<SettingsProps> = ({
   settings,
 }) => {
+  const { binding } = useRuntime()
   const [shippingFreePercentage, setShippingFreePercentage] = useState(0)
   const [differenceBetwenValues, setDifferenceBetwenValues] = useState(0)
+  const [freeShippingAmount, setFreeShippingAmount] = useState(0)
   const {
     orderForm: { totalizers },
   } = useOrderForm()
 
+  useEffect(() => {
+    if (settings.isMultiBindingStore) {
+      const findAmountForBinding = settings.freeShippingAmounts.find(item => item.bindingId === binding?.id)?.freeShippingAmount
+      setFreeShippingAmount(findAmountForBinding ?? 0)
```

Comment:
I might be wrong, but why are we defaulting to `0` here?
Won't this mean that, if a user doesn't set the free shipping amount for every binding of the store, the bindings that are not configured will get free shipping starting from `0`.

If a store is not configured on the app settings, they should either not have free shipping or default to the value set when multi binding is unchecked.

Reply:
You're correct. I adapted the code and tested the following scenarios

1. If freeShippingAmount is set AND bindingBounded is not => use freeShippingAmount -> OK
2. bindingBounded is set AND the freeShippingAmount is set for the current binding => use the freeShippingAmount for that binding -> OK (I had to do the binding check twice 2 determine whether or not it is set, otherwise it would get to the main logic and would cause the amount to be set as 0 AKA initial state)

The reason for the if on line 46 is because find can return undefined as well and react doesn't like that. 

I've tested several other scenarios and everything seems to be covered.